### PR TITLE
[cpp] Immanence logic cleanup

### PR DESCRIPTION
**Pull request by MowFord**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

image

Apologies for the PR re-creation that will be happening, but any of my older PRs that need changes will have to be re-made as I've lost access to the source branches.

I will try to do my best to re-create the comments to avoid extra work on LSB's side.

(I won't be putting this disclaimer on every new PR, just a link to the old)

https://github.com/LandSandBoat/server/pull/5525

## Steps to test these changes
image

Good call on extending the duration, as the skillchain tier does affect the remaining duration of the skillchain effect.

Same testing behavior is useful: add prints to `effects/skillchain.lua` to get the explicit start/stop of the effect (Note that the effect isn't removed when a skillchain is created, so this below screenshot is the logs from the above skillchains)
image

and helixes
image

image

### CODE REVIEW

File: src/map/entities/charentity.cpp [hunk 1]
```diff
@@ -1212,6 +1211,13 @@ void CCharEntity::OnCastFinished(CMagicState& state, action_t& action)
                         actionTarget.addEffectMessage = 287 + effect;
                     }
                     actionTarget.additionalEffect = effect;
+
+                    // Closing a skillchain with an immanence Helix will make the magic burst window longer
+                    auto scEffect = PTarget->StatusEffectContainer->GetStatusEffect(EFFECT_SKILLCHAIN, 0);
+                    if (isHelix && scEffect)
+                    {
+                        scEffect->SetDuration(scEffect->GetDuration() + 2000);
```
Comment (MowFord) [inline src/map/entities/charentity.cpp:1219]: > [WinterSolstice8](https://github.com/WinterSolstice8) [2 weeks ago](https://github.com/LandSandBoat/server/pull/5525#discussion_r1581879360) • 
Second thought: should this not get the duration then add 2000? Skillchains are supposed to have reduced duration at least for following skillchains.

**Comment by CatsEyeXI** [comment]
Sorry for the inconvenience, I tried reaching out to you to discuss this, but never got a reply.
image

**Comment by MowFord** [comment]
Even without being deleted, I wouldn't have been able to make changes and push them. 

Suppose I should have put this screenshot instead:
image

Anyways... apologies again to LSB for having to deal with the duplicated effort

**Comment by zach2good** [comment]
Previous PRs for reference:
https://github.com/LandSandBoat/server/pull/5525
https://github.com/LandSandBoat/server/pull/5273
https://github.com/LandSandBoat/server/pull/5188
https://github.com/LandSandBoat/server/pull/5033

File: src/map/entities/charentity.cpp (hunk 1 above)
Comment (zach2good) [reply src/map/entities/charentity.cpp:1219]: Is this a completely flat 2s bonus, or does it scale with stats/gear/something else? I know _nothing_ about SCH, and this isn't listed obviously when I google `ffxi immanence`

**Review by zach2good — APPROVED** [review]
Code seems fine, just asking about the game logic 👍

File: src/map/entities/charentity.cpp (hunk 1 above)
Comment (MowFord) [reply src/map/entities/charentity.cpp:1219]: Yep. Had it confirmed when I made the PR

https://www.bg-wiki.com/index.php?title=Immanence&diff=prev&oldid=698295

Comment (zach2good) [reply src/map/entities/charentity.cpp:1219]: Delightful, thanks 👍

**Review by zach2good — APPROVED** [review]

**Pull request merged**